common/src/main/scala/uk/ac/wellcome/s3/S3Uri.scala

```diff
@@ -0,0 +1,14 @@
+package uk.ac.wellcome.s3
+
+import java.net.URI
+
+object S3Uri {
```

Comment:
Why use a URI over having a mechanism to serialise/deserialise: https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/S3Object.html?

Reply:
There's also https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3URI.html that more conveniently allows us to extract bucket/key from a URI.

Reply:
Perhaps actually using a plain URI here is ok - as we can use `AmazonS3URI` the other side to extract the bucket and key, making the sending side know a bit less about the storage medium.